topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+====
+{product-long} is currently available for Development Preview. Development Preview releases provide early access to a limited set of features that might not be fully tested and that might change in the final GA version. Users should not use Development Preview software in production or for business-critical workloads. Limited documentation is available for Development Preview releases and is typically focused on fundamental user goals.
+====
+
+// Purpose statement for the assembly
+[role="_abstract"]
+As a developer of applications and services, you can refer to the properties of your topics in {product} to better understand how the Kafka implementation is managed for your services. You can edit certain topic properties according to the needs and goals of your services.
+
+//Additional line break to resolve mod docs generation error, not sure why. Leaving for now. (Stetson, 20 May 2021)
+
+[id="proc-editing-topic-properties_{context}"]
+== Reviewing and editing topic properties in {product}
+
+[role="_abstract"]
+Use the {product} web console to select a topic in your Kafka instance and review the topic properties. You can adjust the editable topic properties as needed.
+
+As an alternative to using the {product} web console, you can use the `rhoas` command-line interface (CLI) to update certain topic properties, as shown in the following example command:
+
+.Example CLI command to update topic retention time
+[source]
+----
+rhoas kafka topic update my-kafka-topic --retention-ms 704800000
+----
+
+For a list of topic properties that you can update using the CLI, see the `rhoas kafka topic update` entry in the {base-url-cli}{rhoas-cli-ref-url}[_CLI command reference (rhoas)_^].
+
+.Prerequisites
+* You are logged in to the {product} web console.
+* You have created a Kafka instance with at least one Kafka topic in {product}.
+
+.Procedure
+. In the {product} web console, go to *Streams for Apache Kafka* > *Kafka Instances* and click the name of the Kafka instance that contains the topics that you want to configure.
+. In the *Topics* page, click the name of the topic that you want to review or edit.
++
+Alternatively, select the options icon (three vertical dots) for the relevant topic and click *Edit*.
+. Click *Edit properties*, review the current topic properties, and adjust any editable topic properties as needed.
+. Click *Save* to finish.
+
+[role="_additional-resources"]
+.Additional resources
+* {base-url}{getting-started-url}[_Getting started with {product}_^]
+* {base-url}{rhoas-cli-url}[_Getting started with the `rhoas` CLI_^]
+* {base-url-cli}{rhoas-cli-ref-url}[_CLI command reference (rhoas)_^]
+
+
+[id="ref-supported-topic-properties_{context}"]
+== Supported topic properties in {product}
+
+[role="_abstract"]
+The following topic properties are supported in {product}. Each listed topic property indicates whether the property is editable or read only, and includes other relevant property attributes for your reference.
```

Comment:
What do we mean by supported here? Will there be other properties that are set implicitly and managed directly by App Services that we don't see. _Topic properties configured in App Services_?

Reply:
So Kafka offers a ton of [topic properties](https://kafka.apache.org/documentation/#topicconfigs). Our product doesn't support all of them, just supports today what is listed in the UI/CLI/this doc. So by "support" we mean literally that, it supports the following list of available Kafka topic properties. To clarify, I've added "following **Kafka** topic properties. That help?

Reply:
Helps. Thanks Stetson